model_compression_toolkit/pytorch/back2framework/model_builder.py

```diff
@@ -0,0 +1,196 @@
+        out_tensors_of_n = input_tensors
+    else:
+        input_tensors = [tensor for tensor_list in input_tensors for tensor in tensor_list]  # flat list of lists
+        op_call_args = n.op_call_args if isinstance(n, FunctionalNode) else []
+        functional_kwargs = n.op_call_kwargs if isinstance(n, FunctionalNode) else {}
+        if isinstance(n, FunctionalNode) and n.inputs_as_list:
+            out_tensors_of_n = op_func(input_tensors, *op_call_args, **functional_kwargs)
+        else:
+            out_tensors_of_n = op_func(*input_tensors + op_call_args, **functional_kwargs)
+
+    # Add a fake quant node if the node has an activation threshold.
+    if mode == ModelBuilderMode.QUANTIZED and n.is_activation_quantization_enabled():
+        fake_quant = n.activation_quantization_cfg.generate_quantization_node()
+
+        if fake_quant is None:
+            raise Exception('Layer is meant to be quantized but fake_quant function is None')
+        out_tensors_of_n = fake_quant(out_tensors_of_n)
+
+    return out_tensors_of_n
+
+
+class PytorchModelBuilder(torch.nn.Module):
+    """
+    Class for reconstructing a Pytorch model from a graph
+    """
+    def __init__(self, graph: Graph,
+                 mode: ModelBuilderMode = ModelBuilderMode.QUANTIZED,
+                 append2output: List[Any] = None):
+        """
+        Construct a Pytorch model.
+        Args:
+            graph: Graph to build its corresponding Pytorch model.
+            mode: Building mode. Read ModelBuilderMode description for more info.
+            append2output: List of nodes or OutTensor objects.
+        """
+        super(PytorchModelBuilder, self).__init__()
+        self.graph = graph
+        self.mode = mode
+        self.node_sort = list(topological_sort(graph))
+        self.nodes_dict = {}
+        self.append2output = append2output
+        for n in self.node_sort:
+            if inspect.isclass(n.layer_class) and issubclass(n.layer_class, torch.nn.Module):
+                self.add_module(n.name, node_builder(n))
+                # setattr(self, n.name, node_builder(n))
+
+    def forward(self,
+                *args: Any) -> Any:
+        """
+
+        Args:
+            inputs: list of input tensors to model
+
+        Returns:
+            torch Tensor/s which is/are the output of the model logic.
+
+        """
+        node_to_output_tensors_dict = dict()
+        for n in self.node_sort:
+            input_tensors = build_input_tensors_list(n,
+                                                     self.graph,
+                                                     args,
+                                                     node_to_output_tensors_dict)
+            out_tensors_of_n = run_operation(n,  # Run node operation and fetch outputs
+                                             input_tensors,
+                                             op_func=getattr(self, n.name) if hasattr(self, n.name) else n.layer_class,
```

Comment:
Also change to:  "n.function_op   if  isinstance(n, FunctionalNode) else getattr(self, n.name)"